### PR TITLE
Expose Jib offline mode

### PR DESCRIPTION
**Pull request by famod**
My company's Artifactory seems to have some spontanous PTO as well and this helped me out.

### CODE REVIEW

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java [hunk 1]
```diff
@@ -153,4 +153,10 @@
      */
     @ConfigItem(defaultValue = "jib-image.id")
     public String imageIdFile;
+
+    /**
+     * Whether or not to operate offline.
+     */
+    @ConfigItem
```
Comment (machi1990) [inline extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:160]: is the default false?

**Review by machi1990 — COMMENTED** [review]
lgtm

@geoand mind having a look?

**Comment by geoand** [comment]
I'll have a look when I get back.

I have no idea what this setting actually does, so I'd have to look at the job docs / code

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java (hunk 1 above)
Comment (famod) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java:160]: I think so. I actually had a look at an existing flag in the same area: https://github.com/quarkusio/quarkus/blob/main/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java#L66

But it's probably better to make it explicit like here: https://github.com/quarkusio/quarkus/blob/main/extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java#L124

I'll push an update.

**Comment by famod** [comment]
Jib seems to check for newer baser image layers by default and that fails if your remote registry is unavailable.
With this flag you can bypass that.

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java (hunk 1 above)
Comment (machi1990) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java]: Just to be clear, I was asking about the default JIB behaviour is this flag is not. If it true, it probably make sense to make our default `true` too, to not "break" existing setup.

Comment (famod) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java]: Ah sorry. It's disabled by default.

Comment (machi1990) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibConfig.java]: Perfect, thanks.

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**